tools/sql/migrate_check.py

```
@@ -0,0 +1,310 @@
+
+
+def go_migrate(dsn: str, *args: str) -> str:
+    r = sh(["go", "run", "./tools/migrate", "-dsn", dsn, *args])
+    if r.returncode != 0:
+        sys.stderr.write(r.stdout + r.stderr)
+        raise SystemExit(f"❌ go migrate {' '.join(args)} 失败")
+    return r.stdout
+
+
+def _normalize_dump(text: str) -> list[str]:
+    """归一化 pg_dump 输出为可比较的语句多重集（排序后比较）。
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)